src/NHibernate/Engine/Query/ParameterParser.cs

```diff
@@ -48,47 +48,47 @@ public static void Parse(string sqlString, IRecognizer recognizer)
 
 			int stringLength = sqlString.Length;
 			bool inQuote = false;
-			bool afterNewLine = false;
 			for (int indx = 0; indx < stringLength; indx++)
 			{
 				int currentNewLineLength;
 
-				// check comments
-				if (indx + 1 < stringLength && sqlString.Substring(indx,2) == "/*")
+				// check comments, unless in quote
+				if (!inQuote)
```

Comment:
We do this look ahead up to two times, it may be worth introducing a local here and guarding furthermore the test.

```suggestion
				// check comments, unless in quote or at end of string
				if (!inQuote && indx + 1 < stringLength)
```